.github/scripts/e2e_tests_metrics.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+set -e
+
+LOG_FILE=".github/e2e_tests/artifacts/logs/e2e_tests/metrics_service.log"
+mkdir -p "$(dirname "$LOG_FILE")"
+
+NAMESPACE="scality-object-storage"
+SERVICE="scality-cosi-driver-metrics"
+LOCAL_PORT=8080
+TARGET_PORT=8080
+
+# Declare expected values for each metric as environment variables
+EXPECTED_CREATE_BUCKET=${1:-0}
+EXPECTED_DELETE_BUCKET=${2:-0}
+EXPECTED_GET_INFO=${3:-1}
+EXPECTED_GRANT_ACCESS=${4:-0}
+EXPECTED_REVOKE_ACCESS=${5:-0}
+GRPC_METHOD_TO_TEST="grpc_server_msg_sent_total"
+
+# Error handling function
+error_handler() {
+  echo "An error occurred during the metrics test. Check the log file for details." | tee -a "$LOG_FILE"
+  echo "Failed command: $BASH_COMMAND" | tee -a "$LOG_FILE"
+  exit 1
+}
+
+# Trap errors and call the error handler
+trap 'error_handler' ERR
+
+# Logging and command execution function
+log_and_run() {
+  echo "Running: $*" | tee -a "$LOG_FILE"
```

Comment:
Not sure I commented this already, but wouldn't a `set -x` be simpler to print command as well ?

Then the whole script could be `tee` into a logfile

Reply:
I tried, it seems to evaluate
```
echo $A
+ echo 9
9
```

But maybe it does something different, anyways it's ok for me